torchrec/distributed/planner/types.py

```diff
@@ -99,9 +101,15 @@ def __init__(
                 )
             )
 
-        self._local_world_size: int = (
-            local_world_size if local_world_size else world_size
-        )
+        if local_world_size is None:
+            try:
+                local_world_size = int(os.getenv("LOCAL_WORLD_SIZE"))
```

Comment:
It seems `torch.distributed.run` will set `LOCAL_WORLD_SIZE` by default.

https://github.com/pytorch/pytorch/blob/b311f255d84b0e745f136117ac21ee2d4f124382/torch/distributed/run.py#L195

So, make local_world_size set to this env variable by default.